source/agora/node/LederOutputStream.d

```diff
@@ -0,0 +1,130 @@
+/*******************************************************************************
+
+    Function definition and helper related to serialization
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.node.LedgerOutputStream;
+
+import agora.common.Transaction;
+
+/// Ditto
+public class LedgerOutputStream
+{
+    private ubyte[] data;
+
+    private size_t offset;
+
+    /// Ctor
+    public this () @trusted
+    {
+        data.length = 0;
+        offset = 0;
+    }
+
+    // Offset must be less than or equal to the length of the data.
+    invariant()
+    {
+        assert(offset <= data.length);
+    }
+
+    // Returns the data from start to offset.
+    ubyte[] toBytes() @trusted
+    {
+        return data[0 .. offset];
+    }
+
+    // Allocate the data size by nbytes.
+    void reserve(size_t nbytes) @trusted
+    in
+    {
+        assert(offset + nbytes >= offset);
+    }
+    out
+    {
+        assert(offset + nbytes <= data.length);
+    }
+    body
+    {
+        if(data.length < offset + nbytes)
+        {
+            void[] vdata = data;
+            vdata.length = (offset + nbytes);
+            data = cast(ubyte[]) vdata;
+        }
+    }
+
+    // Assign additional data size bytes length and store bytes in data
+    // and add offset location bytes size.
+    void write(in ubyte[] bytes) @trusted
+    {
+        reserve(bytes.length);
+        data[offset .. offset + bytes.length] = bytes[];
```

Comment:
```suggestion
        this.data ~= bytes;
```
You can use this instead, and remove the `reserve` call, to achieve the same results.

Reply:
I use this it.